src/fios_gui.cpp

```diff
@@ -802,6 +786,35 @@ struct SaveLoadWindow : public Window {
 		this->vscroll->SetCapacityFromWidget(this, WID_SL_DRIVES_DIRECTORIES_LIST);
 	}
 
+	void BuildDisplayList()
+	{
+		/* Filter changes */
+		this->display_list.clear();
+		this->display_list.reserve(this->fios_items.size());
+
+		if (this->string_filter.IsEmpty()) {
+			/* We don't filter anything out if the filter editbox is empty */
+			for (auto &it : this->fios_items) {
+				this->display_list.push_back(&it);
+			}
+		} else {
+			for (auto &it : this->fios_items) {
+				this->string_filter.ResetState();
+				this->string_filter.AddLine(it.title.c_str());
```

Comment:
```suggestion
				this->string_filter.AddLine(it.title);
```
IIRC the `StringFilter` support `std::string` for `AddLine`.

Reply:
Heh, in a way, yes... `AddLine(str.c_str());`